File: batchbhavna/typesofAdapter/app/src/main/java/com/example/typesofadapter/adapters/CustomSimpleAdapter.java
package com.example.typesofadapter.adapters;

import android.content.Context;
import android.content.Intent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.SimpleAdapter;
import android.widget.Toast;

import com.example.typesofadapter.R;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class CustomSimpleAdapter extends SimpleAdapter {
   ArrayList<HashMap<String,String>> userlist;
   Context context;
    public CustomSimpleAdapter(Context context,ArrayList<HashMap<String,String>> data, int resource, String[] from, int[] to) {
        super(context, data, resource, from, to);
        userlist = data;
        this.context = context;
    }


    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View view =  super.getView(position, convertView, parent);

        Button button = view.findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                show(position);
            }
        });
        return view;
    }

    @Override
    public int getCount() {
        return userlist.size();
    }

    private void show(int a)
    {
        switch (a)
        {
            case 1:
                Toast.makeText(context, "January", Toast.LENGTH_SHORT).show();
                break;
            case 3:
                Toast.makeText(context, "March", Toast.LENGTH_SHORT).show();
                break;
            case 5:
                Toast.makeText(context, "April", Toast.LENGTH_SHORT).show();
                break;
            case 4:
                Toast.makeText(context, "December", Toast.LENGTH_SHORT).show();
                break;
            default:
                Toast.makeText(context, "This is not a case", Toast.LENGTH_SHORT).show();

        }
    }
}

File: 4pmclass/TypesofAdapters/app/src/main/java/com/example/typesofadapters/apdatper/CustomSimpleAdapter.kt
package com.example.typesofadapters.apdatper

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.SimpleAdapter
import android.widget.Toast
import com.example.typesofadapters.R

class CustomSimpleAdapter(var context: Context,var arrylist:ArrayList<HashMap<String,String>>,var resourc:Int,from:Array<String> ,
                          to : IntArray) : SimpleAdapter(context,arrylist,resourc,from,to){
    override fun getCount(): Int {
        return arrylist.size
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {

        var view =super.getView(position, convertView, parent)
            var button =  view.findViewById<Button>(R.id.button)
        button.setOnClickListener {
            sum(position,position)
        }
        return  view;
    }
    fun sum(i:Int,j:Int)
    {
        Toast.makeText(context,"sum is ${i+j}",Toast.LENGTH_SHORT).show()
    }
}